packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -72,6 +72,35 @@ interface CleanTaskDescription {
 
 type TaskDescription = DeliveryTaskDescription | PatrolTaskDescription | CleanTaskDescription;
 
+const isDeliveryTaskDescriptionValid = (taskDescription: DeliveryTaskDescription): boolean => {
+  return (
+    taskDescription.pickup.place.length !== 0 &&
+    taskDescription.pickup.handler.length !== 0 &&
+    taskDescription.pickup.payload.sku.length !== 0 &&
+    taskDescription.pickup.payload.quantity > 0 &&
+    taskDescription.dropoff.place.length !== 0 &&
+    taskDescription.dropoff.handler.length !== 0 &&
+    taskDescription.dropoff.payload.sku.length !== 0 &&
+    taskDescription.dropoff.payload.quantity > 0
+  );
+};
```

Comment:
```tsx
interface Payload {
  sku: string;
  quantity: number;
}
interface TaskPlace {
  place: string;
  handler: string;
  payload: Payload;
}
interface DeliveryTaskDescription {
  pickup: TaskPlace;
  dropoff: TaskPlace;
}
```
DeliveryTaskDescription (line 54) is composed of interfaces instead of being one large object

Reply:
thanks for the suggestion! https://github.com/open-rmf/rmf-web/pull/764/commits/dbc681bd44641943fbd9925e8dd7d77b8c790eb0